prisma/schema.prisma

```diff
@@ -23,6 +23,74 @@ datasource db {
+  metrics                       String[]                  @default([])
+  templateFor                   template_for_enum
+  // Groups defines the "final-achievement"
+  group                         String                    @unique @db.VarChar
+  // Group order tells in which place a subordinate step / achievement is
+  groupOrder                    Int                      
+  // the final achievement is marked with isLast 
+  isLast                        Boolean
+  // what type of achievement is it
+  type                          achievement_type_enum
+  name                          String                    @db.VarChar
+  subtitle                      String                    @db.VarChar
+  description                   String                    @db.VarChar
+  image                         String                    @db.VarChar
+  // some achievements show actions that a user can/must perform next 
+  actionDescription             String?                   @db.VarChar
+  actionRedirectLink            String?                   @db.VarChar
+  actionType                    action_type_enum?
+  // defines what value must be reached to receive the achievement 
+  condition                     String                    @db.VarChar
+  // defined how values must be aggregated to evaluate the condition
+  conditionDataAggregations     Json                      @db.Json
+  isActive                      Boolean 
+}
+model user_achievement {
+  id                            Int                       @id @default(autoincrement())
+  metrics                       String[]                  @default([])
+  templateFor                   template_for_enum
+  group                         String                    @unique @db.VarChar
+  groupOrder                    Int                       
+  isLast                        Boolean
+  type                          achievement_type_enum
+  name                          String                    @db.VarChar
+  subtitle                      String                    @db.VarChar
+  description                   String                    @db.VarChar
+  image                         String                    @db.VarChar
+  actionDescription             String                    @db.VarChar
+  actionRedirectLink            String                    @db.VarChar
+  actionType                    action_type_enum
+  condition                     String                    @db.VarChar
+  conditionDataAggregations     Json                      @db.Json
+  isActive                      Boolean 
+  isSeen                        Boolean
+  isAchieved                    Boolean
+  achievedAt                    DateTime                  @db.Timestamp(6)
+}
+
+// Save (tracked) event as raw data in DB and evaluate at runtime
+model event {
+    id                            Int                       @id @default(autoincrement())
+    userId                        String                    @db.VarChar
```

Comment:
This should be a foreign key relation.

Reply:
it should be, but unfortunately we do not have a "user entity". So I think this is fine, and in line how we do it at other places (concrete_notification, appointment)

Reply:
Ahhh, right. Forgot about that 😅